lite/examples/object_detection/android/app/src/main/java/org/tensorflow/lite/examples/detection/NoteTakingActivity.java

```diff
@@ -0,0 +1,375 @@
+                        final long startTime = SystemClock.uptimeMillis();
+                        final List<Classifier.Recognition> results = detector.recognizeImage(croppedBitmap);
+                        lastProcessingTimeMs = SystemClock.uptimeMillis() - startTime;
+
+                        cropCopyBitmap = Bitmap.createBitmap(croppedBitmap);
+                        final Canvas canvas = new Canvas(cropCopyBitmap);
+                        final Paint paint = new Paint();
+                        paint.setColor(Color.RED);
+                        paint.setStyle(Paint.Style.STROKE);
+                        paint.setStrokeWidth(2.0f);
+
+                        float minimumConfidence = MINIMUM_CONFIDENCE_TF_OD_API;
+                        final List<Classifier.Recognition> mappedRecognitions =
+                                new LinkedList<Classifier.Recognition>();
+
+                        for (final Classifier.Recognition result : results) {
+                            final RectF location = result.getLocation();
+                            if (location != null && result.getConfidence() >= minimumConfidence) {
+                                showPrediction(result.getTitle());
+                            }
+                        }
+
+                        tracker.trackResults(mappedRecognitions, currTimestamp);
+                        trackingOverlay.postInvalidate();
+
+                        computingDetection = false;
+
+                        runOnUiThread(
+                                new Runnable() {
+                                    @Override
+                                    public void run() {
+                                        showCropInfo(cropCopyBitmap.getWidth() + "x" + cropCopyBitmap.getHeight());
+                                    }
+                                });
+                    }
+                });
+    }
+
+    @Override
+    protected int getLayoutId() {
+        return R.layout.tfe_od_camera_connection_fragment_tracking;
+    }
+
+    public int getSquareCropDimensionForBitmap(Bitmap bitmap)
+    {
+        //use the smallest dimension of the image to crop to
+        return Math.min(bitmap.getWidth(), bitmap.getHeight());
+    }
+
+    @Override
+    protected Size getDesiredPreviewFrameSize() {
+        return DESIRED_PREVIEW_SIZE;
+    }
+
+    @Override
+    public void onClick(View view) {
+
+    }
+
+    // Which detection model to use: by default uses Tensorflow Object Detection API frozen
+    // checkpoints.
+    private enum DetectorMode {
+        TF_OD_API;
+    }
+
+    @Override
+    protected void setUseNNAPI(final boolean isChecked) {
+        runInBackground(() -> detector.setUseNNAPI(isChecked));
+    }
+
+    @Override
+    protected void setNumThreads(final int numThreads) {
+        runInBackground(() -> detector.setNumThreads(numThreads));
+    }
+
+    protected void showPrediction(String pred){
```

Comment:
Are you checking for percentage results as well or just letter reoccurrence?

Reply:
Yes. There is a threshold set to 0.9